Mathlib/NumberTheory/LSeries/Basic.lean

```diff
@@ -160,73 +236,103 @@ lemma LSeriesSummable_of_le_const_mul_rpow {f : ArithmeticFunction ℂ} {x : ℝ
       ring
   refine Summable.of_norm <| hsum.of_nonneg_of_le (fun _ ↦ norm_nonneg _) (fun n ↦ ?_)
   rcases n.eq_zero_or_pos with rfl | hn
-  · simp only [map_zero, zero_div, norm_zero, norm_nonneg]
+  · simp only [term_zero, norm_zero]
+    exact norm_nonneg _
   have hn' : 0 < (n : ℝ) ^ s.re := Real.rpow_pos_of_pos (Nat.cast_pos.mpr hn) _
-  simp_rw [norm_div, norm_natCast_cpow_of_pos hn, div_le_iff hn', add_re, sub_re, one_re,
-    ofReal_re, Real.rpow_add <| Nat.cast_pos.mpr hn, div_eq_mul_inv, mul_inv]
-  rw [mul_assoc, mul_comm _ ((n : ℝ) ^ s.re), ← mul_assoc ((n : ℝ) ^ s.re), mul_inv_cancel hn'.ne',
-    ← Real.rpow_neg n.cast_nonneg, norm_eq_abs (C : ℂ), abs_ofReal, _root_.abs_of_nonneg hC₀,
-    neg_sub, one_mul]
-  exact hC n
+  simp_rw [term_of_ne_zero hn.ne', norm_div, norm_natCast_cpow_of_pos hn, div_le_iff hn',
+    norm_eq_abs (C : ℂ), abs_ofReal, _root_.abs_of_nonneg hC₀, div_eq_mul_inv, mul_assoc,
+    ← Real.rpow_neg <| Nat.cast_nonneg _, ← Real.rpow_add <| Nat.cast_pos.mpr hn]
+  simp only [add_re, sub_re, one_re, ofReal_re, neg_add_rev, neg_sub, neg_add_cancel_right]
+  exact hC n <| Nat.pos_iff_ne_zero.mp hn
 
 open Filter Finset Real Nat in
-lemma LSeriesSummable_of_isBigO_rpow {f : ArithmeticFunction ℂ} {x : ℝ} {s : ℂ}
-    (hs : x < s.re) (h : f =O[atTop] fun n ↦ (n : ℝ) ^ (x - 1)) :
+/-- If `f = O(n^(x-1))` and `re s > x`, then the `LSeries` of `f` is summable at `s`. -/
```

Comment:
The proof gives a uniform summability that will be important to prove that the LSeries is holomorphic in s

Reply:
We don't actually need that (I have a proof of the fact that the L-series is (complex) differentiable on its half-plane of absolute convergence via `hasDerivAt_tsum_of_isPreconnected`).

What might be interesting eventually is to also consider conditional convergence. Then it is likely to become relevant for proving that the function is holomorphic on the half-plane of conditional convergence. But this is for future work.

Reply:
Yes, and it will be time, then, to write the lemmas that are necessary.

Reply:
The proof is more delicate, IIRC, involving partial summation, so this will require some work. Any volunteers :smile: ?

Reply:
I vote that we leave issues of conditional convergence for later work – I think such statements would be hard to even formulate in Mathlib as of now, since in mathlib convergence of sums is defined to mean unconditional convergence.

Reply:
We now have the Abel Limit Theorem, which talks about conditional convergence, but so far, it needs to be written as `Tendsto (fun n => (range n).sum f) a` (with `open Filter Finset`). This is workable, but a dedicated API would be nice.